test/functional/p2p_blocksonly.py

```diff
@@ -55,8 +55,29 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
             self.nodes[0].sendrawtransaction(sigtx)
             self.nodes[0].p2p.wait_for_tx(txid)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid))
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
+        self.log.info("Restarting node 0 with whitelist permission and blocksonly")
+        self.restart_node(0, ["-persistmempool=0", "-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
+        self.log.info('Check that txs from whitelisted peers are not rejected and relayed to others')
+        self.nodes[0].add_p2p_connection(P2PInterface())
+        peerInfo = self.nodes[0].getpeerinfo()[0]
+        assert_equal(peerInfo['whitelisted'], True)
+        assert_equal(peerInfo['permissions'], ['noban', 'forcerelay', 'relay', 'mempool']) #checkpermission use check permission from p2p_permission
+
+        self.nodes[0].p2p.send_message(msg_tx(FromHex(CTransaction(), sigtx)))
+
+        pprint(self.nodes[0].p2p.last_message)
+        pprint(self.nodes[0].p2p.last_message.get('tx'))
+
+        assert_equal(self.nodes[0].p2p.is_connected, True)
+        with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=0'.format(txid)]):
+            self.nodes[0].sendrawtransaction(sigtx)
```

Comment:
You will need at least three (mini)nodes if you want to check what you claim to be wanting to check. "Check that txs from whitelisted peers are not rejected and relayed to others"

whitlisted peer --- tx ---> Bitcoin Core ---- tx -----> another peer

```suggestion
            another_peer = self.nodes[0].add_p2p_connection(P2PInterface())
            ... etc
```

Reply:
Achso! I **thnk** I know why I am erroring...I couldn't figure out why `self.nodes[0].p2p.wait_for_tx(txid)` timed out...but little did I realized that  this tx was originated by `self.nodes[0].p2p` hence it wouldn't hear this transaction again?? And instead, `another_peer.wait_for_tx(txid)` would be the correct thing to check for..?

Reply:
> And instead, another_peer.wait_for_tx(txid) would be the correct thing to check for..?

Correct